### PR TITLE
Add appModule for Spring Tool Suite 4

### LINKED ISSUE #10001: redundant speech when navigating trees in eclipse
### Steps to reproduce:
1. open eclipse and press alt+shift+q p to go to package explorer
2. use arrows to navigate the tree
### Actual behavior:
nvda reads the item it is leaving and the item that got focus 
### Expected behavior:
read only the item that got focus
#### NVDA installed/portable/running from source:
nvda installed
#### NVDA version:
nvda 2019.2.rc1
#### Windows version:
windows10
#### Name and version of other software in use when reproducing the issue:
spring tool 4
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
it also happens in version 2019.1 
I guess some thing changed in windows

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Fixes #10001 
### Summary of the issue:
Spring Tool Suite 4 changed name of its main executable. Because of this proper appModule wasn't loaded.
### Description of how this pull request fixes the issue:
Add app Module for a proper executable. This simply imports module for Eclipse.
### Testing performed:
Try build tested by @alexandretoco. He confirms that proper module is loaded. I cannot test myself because Spring Tool suite fails to start on my machine for some reason.
### Known issues with pull request:
None known
### Change log entry:

Section: Bug fixes

Spring Tool Suite Version 4 is now supported

### CODE REVIEW

**Comment by lukaszgo1** [comment]
This PR causes two Flake8 errors: F403 and F401. Can I simply ignore these with an appropriate noqa? All app modules importing module for different program are doing exactly the same.  cc @feerrenrut

**Comment by lukaszgo1** [comment]
@alexandretoco Could you please download [this build](https://ci.appveyor.com/api/buildjobs/n1red402sa9r9gk0/artifacts/output%2Fnvda_snapshot_pr10339-18879%2C1566466d.exe), temporarily remove module that I've send you privately from the scratchpad directory and confirt that the issue is fixed?

**Comment by alexandretoco** [comment]
Hi Lukasz!
I can confirm that the snapshot corrected the problem in eclipse.
How ever, I came back to the 2019.2.1 because I use eloquence and the snapshot didn’t recognize it.


Enviado do Email para Windows 10

De: Łukasz Golonka
Enviado:domingo, 6 de outubro de 2019 16:45
Para: nvaccess/nvda
Cc:alexandretoco; Mention
Assunto: Re: [nvaccess/nvda] Add appModule for Spring Tool Suite 4 (#10339)

@alexandretoco Could you please download this build, temporarily remove module that I've send you privately from the scratchpad directory and confirt that the issue is fixed?

**Comment by lukaszgo1** [comment]
The add-on that you ate using is not compatible with Python 3, so this isn't caused by this PR.

**Comment by feerrenrut** [comment]
>  Can I simply ignore these with an appropriate noqa? 

In most cases, despite it breaking consistency, I would prefer just what is used be imported explicitly. In this case, because the goal is to map a new executable name to the same app module without having to maintain another file then I agree `import *` is the least brittle way. However please make sure you add a comment to explain why this `#noqa` is an exception.

I would really like app modules to be mapped via a more flexible mechanism than by matching the python file name with the executable.

**Comment by lukaszgo1** [comment]
@feerrenrut Done, I hope my commend is clear enough. I'd be nice to have this merged in time for 2019.3 because without this fix latest version of Spring Tool isn't really usable according to @alexandretoco

File: source/appModules/springtoolsuite4.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: UTF-8 -*-
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2019 NV Access Limited, Łukasz Golonka
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+""" App module for Spring Tool Suite version 4
+This simply uses the app module for Eclipse.
+Normally these Flake8 errors shouldn't be ignored but in this case we are simply reusing existing ap module.
```
Comment (leonardder) [inline source/appModules/springtoolsuite4.py]: Could you please move this line out of the doc string and make it a comment?

**Review by LeonarddeR — CHANGES_REQUESTED** [review]

**Comment by leonardder** [comment]
@feerrenrut wrote:
> I would really like app modules to be mapped via a more flexible mechanism than by matching the python file name with the executable.

see #10248, which is related to that desire.

**Comment by lukaszgo1** [comment]
@leonardder You have pending review here.

File: source/appModules/springtoolsuite4.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+""" App module for Spring Tool Suite version 4
+This simply uses the app module for Eclipse.
+"""
+
+# Normally these Flake8 errors shouldn't be ignored but here we are simply reusing existing ap module.
+
+from .eclipse import *  # noqa: F403,F401
```
Comment (leonardder) [inline source/appModules/springtoolsuite4.py]:
```suggestion
from .eclipse import *  # noqa: F401, F403
```

**Review by LeonarddeR — APPROVED** [review]

**Comment by lukaszgo1** [comment]
@feerrenrut This is ready for a review if you have time.

**Comment by lukaszgo1** [comment]
@feerrenrut Would you be able to review this, and if it looks ok get it into 2019.3? I understand that we are quite close to the release, however the change itself is very small and without it someone depending on latest version of Spring Tool has to wait a few more months.

**Review by feerrenrut — APPROVED** [review]
Thanks @lukaszgo1

**Pull request merged**